app/presenters/hyrax/iiif_manifest_presenter.rb

```diff
@@ -0,0 +1,222 @@
+# frozen_string_literal: true
+
+module Hyrax
+  ##
+  # This presenter wraps objects in the interface required by `IIIFManifiest`.
+  # It will accept either a Work-like resource or a SolrDocument.
+  #
+  # @example with a work
+  #
+  #   monograph = Monograph.new
+  #   presenter = IiifManifestPresenter.new(monograph)
+  #   presenter.title # => []
+  #
+  #   monograph.title = ['Comet in Moominland']
+  #   presenter.title # => ['Comet in Moominland']
+  #
+  # @see https://www.rubydoc.info/gems/iiif_manifest
+  class IiifManifestPresenter < Draper::Decorator
+    delegate_all
+
+    ##
+    # @!attribute [w] ability
+    #   @return [Ability]
+    # @!attribute [w] hostname
+    #   @return [String]
+    attr_writer :ability, :hostname
+
+    class << self
+      ##
+      # @param [Hyrax::Resource, SolrDocument]
+      def for(model)
+        klass = model.file_set? ? DisplayImagePresenter : IiifManifestPresenter
```

Comment:
Can I get a bit of an explanation on this method? It's a bit surprising to see the `for` pick this class or another class. It implies that the other class (e.g., DisplayImagePresenter) should have a similar logic switch. Which then leads me to look at the namespace organization of these.

Again, not a blocker, but a surprise.

Reply:
the way i'm thinking of this is `DisplayImagePresenter` is the `IiifManifestPresenter` equivalent for FileSets. this factory method's scope is to give me a presenter for use by `iiif_manifest`, appropriate for the given object.

>  It implies that the other class (e.g., DisplayImagePresenter) should have a similar logic switch. 

can you say more about this?

Reply:
I was assuming that both classes were in the same namespace level and not that the `DisplayImagePresenter` was an inner class. Once I saw it as an inner-class, I'm less concerned about this method. I do believe it highlights that the `IiifManifestPresenter` may in fact make sense to have a namespacing module (e.g. `IiifManifestPresentation`) and move the `IiifManifestPresenter` within that module. Then the `IiifManifestPresentation.for` method, which determines which inner class to initialize.

Reply:
ah yeah. `DisplayImagePresenter` is kept as an inner class with the idea that it provides encapsulation. generally, only `iiif_manifest` callers should be unwrapping enough to see the "Display Image" parts of this, so from Hyrax's perspective `IiifManifestPresenter` does all the work.